File: 1359-count-all-valid-pickup-and-delivery-options/1359-count-all-valid-pickup-and-delivery-options.cpp
class Solution {
public:
    int countOrders(int n) {
        int MOD=1000000007;
        long long int arr[n+1];
        arr[1]=1;
        for(int i=2;i<=n;i++)
        {
            int a=((2*i)-1)%MOD;
            arr[i]=(((arr[i-1])%MOD)*(((a*(a+1))%MOD)/2)%MOD)%MOD;
        }
        return (int)arr[n];
        
    }
};

File: 1359-count-all-valid-pickup-and-delivery-options/1359-count-all-valid-pickup-and-delivery-options.java
class Solution {
    public int countOrders(int n) {
        long [] dp=new  long [n+1];
        long mod=1000000007;
        dp[1]=1;
        for(int i=2;i<=n;i++) {
            int a=2*i-1;
            dp[i]=((dp[i-1]%mod)*(((a*(a+1))/2)%mod))%mod;
        }
        return (int)dp[n];
        
    }
}